lib/app/app_base.nit

```diff
@@ -115,5 +115,19 @@ end
 # The running `App`
 fun app: App do return once new App
 
+# Platform bound at compilation (by importation or -m)
+#
+# This value should not be used to decide the behavior of the software.
+# Class refinement provide a safer and a static solution to apply variations.
+# However, this value can be used in log files and communications with servers.
+fun bound_platform: String do return "none"
+
+if "NIT_TESTING".environ == "true" then exit 0
+
+if bound_platform == "none" then
```

Comment:
Your point is so nicely worded!

Since platform variations are mandatory, by definition, an app without such a variation is not an app.nit app. As such, it can decide its behavior using `bound_platform`.

More seriously, I don't see another way to do this as of now. We could use an annotation to declare mandatory importations which,  if missing, would block compilation.
~~~
module app is mandatory(android or linux)
~~~

Or to use class refinement, we would need negative conditional importations, something like:
~~~
import crash_at_launch is conditional(not android and not linux)
~~~

But even these solutions would require some trick to for unit testing. But I remain open to suggestions!